src/event.rs

```diff
@@ -0,0 +1,616 @@
+where
+	K::Target: KVStorePersister,
+	L::Target: Logger,
+{
+	wallet: Arc<Wallet<bdk::sled::Tree>>,
+	event_queue: Arc<EventQueue<K>>,
+	channel_manager: Arc<ChannelManager>,
+	network_graph: Arc<NetworkGraph>,
+	keys_manager: Arc<KeysManager>,
+	inbound_payments: Arc<PaymentInfoStorage>,
+	outbound_payments: Arc<PaymentInfoStorage>,
+	tokio_runtime: RwLock<Option<Arc<tokio::runtime::Runtime>>>,
+	logger: L,
+	_config: Arc<Config>,
+}
+
+impl<K: Deref, L: Deref> EventHandler<K, L>
+where
+	K::Target: KVStorePersister,
+	L::Target: Logger,
+{
+	pub fn new(
+		wallet: Arc<Wallet<bdk::sled::Tree>>, event_queue: Arc<EventQueue<K>>,
+		channel_manager: Arc<ChannelManager>, network_graph: Arc<NetworkGraph>,
+		keys_manager: Arc<KeysManager>, inbound_payments: Arc<PaymentInfoStorage>,
+		outbound_payments: Arc<PaymentInfoStorage>, logger: L, _config: Arc<Config>,
+	) -> Self {
+		let tokio_runtime = RwLock::new(None);
+		Self {
+			event_queue,
+			wallet,
+			channel_manager,
+			network_graph,
+			keys_manager,
+			inbound_payments,
+			outbound_payments,
+			logger,
+			tokio_runtime,
+			_config,
+		}
+	}
+
+	pub(crate) fn set_runtime(&self, tokio_runtime: Arc<tokio::runtime::Runtime>) {
+		*self.tokio_runtime.write().unwrap() = Some(tokio_runtime);
+	}
+
+	pub(crate) fn drop_runtime(&self) {
+		*self.tokio_runtime.write().unwrap() = None;
+	}
+}
+
+impl<K: Deref, L: Deref> LdkEventHandler for EventHandler<K, L>
+where
+	K::Target: KVStorePersister,
+	L::Target: Logger,
+{
+	fn handle_event(&self, event: LdkEvent) {
+		match event {
+			LdkEvent::FundingGenerationReady {
+				temporary_channel_id,
+				counterparty_node_id,
+				channel_value_satoshis,
+				output_script,
+				..
+			} => {
+				// Construct the raw transaction with one output, that is paid the amount of the
+				// channel.
+				let confirmation_target = ConfirmationTarget::Normal;
+
+				// Sign the final funding transaction and broadcast it.
+				match self.wallet.create_funding_transaction(
+					&output_script,
+					channel_value_satoshis,
+					confirmation_target,
+				) {
+					Ok(final_tx) => {
+						// Give the funding transaction back to LDK for opening the channel.
+						if self
+							.channel_manager
+							.funding_transaction_generated(
+								&temporary_channel_id,
+								&counterparty_node_id,
+								final_tx,
+							)
+							.is_err()
+						{
+							log_error!(self.logger, "Channel went away before we could fund it. The peer disconnected or refused the channel");
+						}
+					}
+					Err(_err) => {
+						self.channel_manager
+							.force_close_without_broadcasting_txn(
+								&temporary_channel_id,
+								&counterparty_node_id,
+							)
+							.expect(
+								"Failed to force close channel after funding generation failed",
+							);
+					}
+				}
+			}
+			LdkEvent::PaymentReceived { payment_hash, purpose, amount_msat } => {
+				log_info!(
+					self.logger,
+					"Received payment from payment hash {} of {} millisatoshis",
```

Comment:
"msat" is used in later on. Probably more consistent with LDK to use "msats" throughout.

Reply:
Done.